File: internal/model/request.go
package model

// SendRequestInput 前端发送请求时传入的参数
type SendRequestInput struct {
	Method  string          `json:"method"`
	URL     string          `json:"url"`
	Params  []KeyValue      `json:"params"`
	Headers []KeyValue      `json:"headers"`
	Body    RequestBody     `json:"body"`
	Auth    AuthConfig      `json:"auth"`
	Options *RequestOptions `json:"options,omitempty"`
}

type RequestOptions struct {
	FollowRedirects         bool   `json:"followRedirects"`
	TimeoutMS               int    `json:"timeoutMs"`
	MaxResponseSizeMB       int    `json:"maxResponseSizeMB"`
	SSLVerify               bool   `json:"sslVerify"`
	HTTPVersion             string `json:"httpVersion"`
	DisableDefaultUserAgent bool   `json:"disableDefaultUserAgent"`
	DisableDefaultAccept    bool   `json:"disableDefaultAccept"`
	DisableAutoContentType  bool   `json:"disableAutoContentType"`
}

type KeyValue struct {
	Key         string `json:"key"`
	Value       string `json:"value"`
	Description string `json:"description,omitempty"`
}

type RequestBody struct {
	Type           string     `json:"type"`
	Raw            string     `json:"raw"`
	JSON           string     `json:"json"`
	FormUrlEncoded []KeyValue `json:"formUrlEncoded"`
	FormData       []FormData `json:"formData"`
}

type FormData struct {
	Key         string `json:"key"`
	Value       string `json:"value"`
	Description string `json:"description,omitempty"`
	Type        string `json:"type"` // text | file
	FilePath    string `json:"filePath,omitempty"`
	FileName    string `json:"fileName,omitempty"`
}

type AuthConfig struct {
	Type   string     `json:"type"`
	Basic  BasicAuth  `json:"basic"`
	Bearer BearerAuth `json:"bearer"`
	APIKey APIKeyAuth `json:"apiKey"`
}

type BasicAuth struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type BearerAuth struct {
	Token string `json:"token"`
}

type APIKeyAuth struct {
	Key   string `json:"key"`
	Value string `json:"value"`
	AddTo string `json:"addTo"`
}

File: frontend/src/types/request.ts
import type { HttpMethod, BodyType, AuthType } from "@/lib/constants"

export interface KeyValuePair {
  id: string
  key: string
  value: string
  enabled: boolean
  description?: string
}

export interface FormDataItem extends KeyValuePair {
  type: "text" | "file"
  filePath?: string
  fileName?: string
}

export interface RequestBody {
  type: BodyType
  raw?: string
  json?: string
  formUrlEncoded?: KeyValuePair[]
  formData?: FormDataItem[]
}

export interface BasicAuthConfig {
  username: string
  password: string
}

export interface BearerAuthConfig {
  token: string
}

export interface ApiKeyAuthConfig {
  key: string
  value: string
  addTo: "header" | "query"
}

export interface AuthConfig {
  type: AuthType
  basic?: BasicAuthConfig
  bearer?: BearerAuthConfig
  apiKey?: ApiKeyAuthConfig
}

export interface RequestData {
  id: string
  name: string
  method: HttpMethod
  url: string
  params: KeyValuePair[]
  headers: KeyValuePair[]
  body: RequestBody
  auth: AuthConfig
  folderId?: string
  projectId?: string
  createdAt: string
  updatedAt: string
}

export function createDefaultRequest(partial?: Partial<RequestData>): RequestData {
  return {
    id: crypto.randomUUID(),
    name: "New Request",
    method: "GET",
    url: "",
    params: [],
    headers: [],
    body: { type: "none" },
    auth: { type: "none" },
    createdAt: new Date().toISOString(),
    updatedAt: new Date().toISOString(),
    ...partial,
  }
}

export function createKeyValuePair(partial?: Partial<KeyValuePair>): KeyValuePair {
  return {
    id: crypto.randomUUID(),
    key: "",
    value: "",
    enabled: true,
    ...partial,
  }
}

export function createFormDataItem(partial?: Partial<FormDataItem>): FormDataItem {
  return {
    id: crypto.randomUUID(),
    key: "",
    value: "",
    enabled: true,
    type: "text",
    ...partial,
  }
}
